src/xdp-utils.c

```diff
@@ -808,6 +856,48 @@ parse_app_info_from_snap (pid_t pid, GError **error)
   return g_steal_pointer (&app_info);
 }
 
+XdpAppInfo *
+xdp_app_info_new_test (pid_t pid,
+                       GError **error)
+{
+  g_autoptr(XdpAppInfo) app_info = NULL;
+  g_autoptr(GKeyFile) key_file = NULL;
+  g_autofree char *test_file_app_info = NULL;
+  g_autofree char *key_data = NULL;
+  size_t key_data_len;
+
+  g_assert (g_get_user_data_dir () != NULL);
+
+  test_file_app_info =
+    g_build_filename (g_get_user_data_dir (), TEST_METADATA_FILE_NAME, NULL);
+
+  if (!g_file_get_contents (test_file_app_info, &key_data, &key_data_len, NULL))
```

Comment:
NULL -> error

Reply:
So, the point here was that the function cal also return `NULL` when there is not an error... Like we accept that the file does not exists, so we don't have a a test-app, but it's not an error.

And the code calling this take care of it (I think other types do the same).